lib/puppet/provider/package/dnfmodule.rb

```diff
@@ -98,6 +99,18 @@ def install
     end
   end
 
+  # should only get here when @resource[ensure] is :disabled
+  def insync?(is)
+    if resource[:ensure] == :disabled
+      # in sync only if package is already disabled
+      pkg = self.class.instances.find do |package|
+        @resource[:name] == package.name && package.properties[:ensure] == :disabled
+      end
+      return true if pkg
+    end
+    return false
+  end
+
```

Comment:
I'm not sure we need this code. I removed it and the functionality still worked.

Reply:
I get some problems only when ensuring disabled.
Running `puppet resource package swig ensure=disabled provider=dnfmodule` without the insync method, I get
```
Error: undefined method `index' for :disabled:Symbol
Error: /Package[swig]/ensure: change from '3.0' to 'disabled' failed: undefined method `index' for :disabled:Symbol
```
I think this is because if ensure is set to :disable, it will try the insync method in provider/package/rpm.rb, where it fails.